src/kernel/capinv.c

```diff
@@ -1442,6 +1444,7 @@ composite_syscall_slowpath(struct pt_regs *regs, int *thd_switch)
 			n = asnd_process(rthd, thd, tcapdst, tcap_current(cos_info), &tcap_next, yield);
 			if (n != thd) {
 				ret = cap_switch(regs, thd, n, tcap_next, TCAP_TIME_NIL, ci, cos_info);
+				if (unlikely(ret < 0)) cos_throw(err, ret);
```

Comment:
`cap_switch` return value at this point is same as `cap_thd_switch` which only returns `0` or `1`. This is mainly for unexpected return value for documentation and readability. (even if we extend cap_switch/cap_thd_switch to return errors, this shouldn't need to be revisited). 

That's the main reason to add this.